drivers/sensor/bmp085/bmp085.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2024, Mario Paja
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bosch_bmp085
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/types.h>
+
+#include "bmp085.h"
+
+LOG_MODULE_REGISTER(BMP085, CONFIG_SENSOR_LOG_LEVEL);
+
+static int bmp085_read_raw_temp(const struct device *dev)
+{
+	struct bmp085_data *data = dev->data;
+	int ret = 0;
+	uint8_t buf[2];
+
+	ret = bmp085_write_byte(dev, BMP085_CTRL, BMP085_CMD_READ_TEMP);
+
+	if (ret < 0) {
+		LOG_DBG("bmp085_write");
+		return ret;
+	}
+	k_msleep(5);
+
+	ret = bmp085_read(dev, BMP085_REG_MSB, buf, sizeof(buf));
+	if (ret < 0) {
+		LOG_DBG("Failed to read MSB.");
+		return ret;
+	}
+
+	data->raw_temp = buf[1] | (buf[0] << 8);
+
+	return ret;
+}
+
+static int bmp085_read_raw_pres(const struct device *dev, uint8_t oversampling)
+{
+	struct bmp085_data *data = dev->data;
+	const struct bmp085_config *cfg = dev->config;
+	int ret = 0;
+	uint8_t buf[2];
+	uint8_t buf_xlsb;
+
+	ret = bmp085_write_byte(dev, BMP085_CTRL, (BMP085_CMD_READ_PRESS + (oversampling << 6)));
+	if (ret < 0) {
+		LOG_DBG("bmp085_write");
+		return ret;
+	}
+
+	if (cfg->oversampling == BMP085_MODE_1_ULTRALOWPOWER) {
+		k_msleep(5);
+	} else if (cfg->oversampling == BMP085_MODE_2_STANDARD) {
+		k_msleep(8);
+	} else if (cfg->oversampling == BMP085_MODE_3_HIGHRES) {
+		k_msleep(14);
+	} else {
+		k_msleep(26);
+	}
+
+	ret = bmp085_read(dev, BMP085_REG_MSB, buf, sizeof(buf));
+	if (ret < 0) {
+		LOG_DBG("Failed to read MSB.");
+		return ret;
+	}
+
+	data->raw_press = buf[1] | (buf[0] << 8);
+	data->raw_press <<= 8;
+
+	ret = bmp085_read(dev, BMP085_REG_XLSB, &buf_xlsb, sizeof(buf_xlsb));
+	if (ret < 0) {
+		LOG_DBG("Failed to read MSB.");
```

Comment:
Just spotted this minor typo, sorry I didn't see it earlier

Reply:
Done :)